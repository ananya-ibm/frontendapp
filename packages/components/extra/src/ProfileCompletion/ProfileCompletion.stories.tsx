/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { ProfileCompletion } from './ProfileCompletion';

export default {
  title: 'Components/Extra/ProfileCompletion',
  component: ProfileCompletion
};

const storyProps = {
  percent: 50,
  items: [
    {
      title: 'Personal details',
      isComplete: true
    },
    {
      title: 'Billing information',
      isComplete: true
    },
    {
      title: 'Security preferences',
      isComplete: false
    }
  ]
};

export const normal = args => <ProfileCompletion {...args} />;
normal.args = storyProps;
