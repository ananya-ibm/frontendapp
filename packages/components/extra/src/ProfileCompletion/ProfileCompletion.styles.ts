/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import styled from 'styled-components';
import { responsiveFontBlock } from '@exo/frontend-common-style-utils';

export const ProfileCompletion = styled('div')`
  display: flex;
  padding: 2rem;
`;

export const Roundel = styled('div')`
  color: ${props => props.theme.colors.information.success};
  flex: 0 0 8rem;
  ${props => responsiveFontBlock(props.theme.typography.heading.heading1)};
  height: 8rem;
  justify-content: center;
  width: 8rem;
`;
export const Details = styled('ul')`
  flex: 0 0 75%;
  list-style: none;
  padding-left: 2rem;
  padding-top: 1rem;

  & .ProfileCompletion-item {
    margin: 1rem 0 0;
  }

  & .ProfileCompletion-icon {
    margin-right: 0.25rem;

    &.isComplete {
      color: ${props => props.theme.colors.information.success};
    }
  }
`;
