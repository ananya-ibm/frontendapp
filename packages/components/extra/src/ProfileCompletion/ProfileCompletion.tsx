/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { Progress } from '../Progress/Progress';
import * as S from './ProfileCompletion.styles';

export const ProfileCompletion = ({ items, percent }: Props) => {
  return (
    <S.ProfileCompletion>
      <S.Roundel>
        <Progress value={percent} />
      </S.Roundel>
      <S.Details>
        {items &&
          items.map(item => (
            <li key={item.title} className="ProfileCompletion-item">
              <span className={`ProfileCompletion-icon ${item.isComplete ? ' isComplete' : ''}`}>
                {item.isComplete ? '✔' : '—'}
              </span>
              {item.title}
            </li>
          ))}
      </S.Details>
    </S.ProfileCompletion>
  );
};

type Props = {
  percent: number;
  items: {
    title: string;
    isComplete: boolean;
  }[];
};
