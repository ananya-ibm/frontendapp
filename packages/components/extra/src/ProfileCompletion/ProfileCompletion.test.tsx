/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';

import { renderWithTheme as render } from '@testUtils';
import { axe } from 'jest-axe';
import { ProfileCompletion } from './ProfileCompletion';

const testProps = {
  percent: 50,
  items: [
    {
      title: 'Personal details',
      isComplete: true
    },
    {
      title: 'Billing information',
      isComplete: true
    },
    {
      title: 'Security preferences',
      isComplete: false
    }
  ]
};

describe('<ProfileCompletion /> component', () => {
  test('its snapshot matches', () => {
    const { container } = render(<ProfileCompletion {...testProps} />);

    expect(container).toBeDefined();
    expect(container.firstChild).toMatchSnapshot();
  });
  test('it passes an axe accesibility check', async () => {
    const { container } = render(<ProfileCompletion {...testProps} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
