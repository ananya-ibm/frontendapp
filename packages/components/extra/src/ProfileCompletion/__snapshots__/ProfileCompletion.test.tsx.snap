// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileCompletion /> component its snapshot matches 1`] = `
.c2 {
  max-height: 20rem;
  max-width: 20rem;
  position: relative;
}

.c2 .CircularProgressbar {
  vertical-align: middle;
  width: 100%;
}

.c2 .CircularProgressbar-path {
  stroke: red;
}

.c2 .CircularProgressbar-trail {
  stroke: gray;
}

.c2 .CircularProgressbar-text {
  fill: yellow;
}

.c2 .CircularProgressbar-background {
  fill: green;
}

.c2 .CircularProgressbar .CircularProgressbar-path {
  stroke: #0062ff;
  stroke-linecap: round;
  -webkit-transition: stroke-dashoffset 0.5s ease 0s;
  transition: stroke-dashoffset 0.5s ease 0s;
}

.c2 .CircularProgressbar .CircularProgressbar-trail {
  stroke: #c6c6c6;
  stroke-linecap: round;
}

.c2 .CircularProgressbar .CircularProgressbar-text {
  -webkit-dominant-baseline: middle;
  dominant-baseline: middle;
  fill: #0062ff;
  font-size: 1.3rem;
  text-anchor: middle;
}

.c2 .CircularProgressbar .CircularProgressbar-background {
  fill: #c6c6c6;
}

.c2 .CircularProgressbar.CircularProgressbar-inverted .CircularProgressbar-background {
  fill: #c9deff;
}

.c2 .CircularProgressbar.CircularProgressbar-inverted .CircularProgressbar-text {
  fill: #ffffff;
}

.c2 .CircularProgressbar.CircularProgressbar-inverted .CircularProgressbar-path {
  stroke: #ffffff;
}

.c2 .CircularProgressbar.CircularProgressbar-inverted .CircularProgressbar-trail {
  stroke: transparent;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 2rem;
}

.c1 {
  color: #198038;
  -webkit-flex: 0 0 8rem;
  -ms-flex: 0 0 8rem;
  flex: 0 0 8rem;
  font: 700 2.625rem/3.1473750000000003rem 'IBM Plex Sans',Arial,sans-serif;
  height: 8rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 8rem;
}

.c3 {
  -webkit-flex: 0 0 75%;
  -ms-flex: 0 0 75%;
  flex: 0 0 75%;
  list-style: none;
  padding-left: 2rem;
  padding-top: 1rem;
}

.c3 .ProfileCompletion-item {
  margin: 1rem 0 0;
}

.c3 .ProfileCompletion-icon {
  margin-right: 0.25rem;
}

.c3 .ProfileCompletion-icon.isComplete {
  color: #198038;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <svg
        class="CircularProgressbar "
        data-test-id="CircularProgressbar"
        viewBox="0 0 100 100"
      >
        <path
          class="CircularProgressbar-trail"
          d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
          fill-opacity="0"
          stroke-width="8"
          style="stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 0px;"
        />
        <path
          class="CircularProgressbar-path"
          d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
          fill-opacity="0"
          stroke-width="8"
          style="stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 144.51326206513048px;"
        />
        <text
          class="CircularProgressbar-text"
          x="50"
          y="50"
        >
          50%
        </text>
      </svg>
    </div>
  </div>
  <ul
    class="c3"
  >
    <li
      class="ProfileCompletion-item"
    >
      <span
        class="ProfileCompletion-icon  isComplete"
      >
        ✔
      </span>
      Personal details
    </li>
    <li
      class="ProfileCompletion-item"
    >
      <span
        class="ProfileCompletion-icon  isComplete"
      >
        ✔
      </span>
      Billing information
    </li>
    <li
      class="ProfileCompletion-item"
    >
      <span
        class="ProfileCompletion-icon "
      >
        —
      </span>
      Security preferences
    </li>
  </ul>
</div>
`;
